goth_tests/domain/ya-provider/test_provider_multi_activity.py

```diff
@@ -159,3 +159,184 @@ async def test_provider_single_simultaneous_activity(
 
         await requestor.terminate_agreement(agreement_id, None)
         await provider.wait_for_agreement_terminated()
+
+
+@pytest.mark.asyncio
+async def test_provider_renegotiate_proposal(
+    common_assets: Path,
+    config_overrides: List[Override],
+    log_dir: Path,
+):
+    """Tests providers' ability of renegotiating previously rejected proposal."""
+
+    nodes = [
+        {"name": "requestor-1", "type": "Requestor"},
+        {"name": "requestor-2", "type": "Requestor"},
+        {"name": "provider-1", "type": "VM-Wasm-Provider", "use-proxy": True},
+    ]
+    config_overrides.append(("nodes", nodes))
+
+    runner, config = _create_runner(common_assets, config_overrides, log_dir)
+
+    async with runner(config.containers):
+        requestor1, requestor2 = runner.get_probes(probe_type=RequestorProbe)
+        providers = runner.get_probes(probe_type=ProviderProbe)
+        assert providers
+
+        def build_demand(requestor):
+            return (
+                DemandBuilder(requestor)
+                .props_from_template(wasi_task_package)
+                .property("golem.srv.caps.multi-activity", True)
+                .constraints(
+                    "(&(golem.com.pricing.model=linear)\
+                    (golem.srv.caps.multi-activity=true)\
+                    (golem.runtime.name=wasmtime))"
+                )
+                .build()
+            )
+
+        async def negotiate_begin(requestor, demand, providers):
+            logger.info(
+                "%s Negotiating with providers",
+                requestor.name,
+            )
+            for provider in providers:
+                await provider.wait_for_offer_subscribed()
+
+            subscription_id, demand = await requestor.subscribe_demand(demand)
+
+            proposals = await requestor.wait_for_proposals(
+                subscription_id,
+                providers,
+                lambda p: p.properties.get("golem.runtime.name") == "wasmtime",
+            )
+            logger.info("Collected %s proposals", len(proposals))
+            return subscription_id, proposals
```

Comment:
we should assert that `proposals` is not empty

Reply:
Afterwards I've realised that `wait_for_proposals` returns the same number of proposals what number of providers is on input.

Reply:
Yes, the length should be equal to number of providers. Altough I don't think it's the focus of this test I'll add an assert.